kata-deploy/scripts/kata-deploy.sh

```diff
@@ -114,29 +134,46 @@ function configure_containerd() {
    [plugins.cri.containerd]
      [plugins.cri.containerd.runtimes.kata]
         runtime_type = "io.containerd.kata.v2"
+        [plugins.cri.containerd.runtimes.kata.options]
+	      ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration.toml"
+     [plugins.cri.containerd.runtimes.kata-fc]
+        runtime_type = "io.containerd.kata-fc.v2"
+        [plugins.cri.containerd.runtimes.kata-fc.options]
+	      ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-fc.toml"
+     [plugins.cri.containerd.runtimes.kata-qemu]
+        runtime_type = "io.containerd.kata-qemu.v2"
+        [plugins.cri.containerd.runtimes.kata-qemu.options]
+	      ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-qemu.toml"
+     [plugins.cri.containerd.runtimes.kata-nemu]
+        runtime_type = "io.containerd.kata-nemu.v2"
+        [plugins.cri.containerd.runtimes.kata-nemu.options]
+	      ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-nemu.toml"
 EOT
-
-
 	#Currently containerd has an assumption on the location of the shimv2 implementation
 	#Until support is added (see https://github.com/containerd/containerd/issues/3073),
-        #create a link in /usr/local/bin/ to the v2-shim implementation in /opt/kata/bin.
-	if [ -f ${shim_file} ]; then
-		echo "warning: ${shim_binary} already exists" >&2
-		if [ ! -f ${shim_backup} ]; then
-			mv ${shim_file} ${shim_backup}
-		else
-			rm ${shim_file}
-		fi
-	fi
+    #create a link in /usr/local/bin/ to the v2-shim implementation in /opt/kata/bin.
 
 	mkdir -p /usr/local/bin
 
-	cat << EOT | tee "$shim_file"
+	for shim in ${shims[@]}; do
+		local shim_binary="containerd-shim-kata-${shim}-v2"
+		local shim_file="/usr/local/bin/${shim_binary}"
+		local shim_backup="/usr/local/bin/${shim_binary}.bak"
+
+		if [ -f ${shim_file} ]; then
+			echo "warning: ${shim_binary} already exists" >&2
+			if [ ! -f ${shim_backup} ]; then
+				mv ${shim_file} ${shim_backup}
+			else
+				rm ${shim_file}
+			fi
+		fi
+       cat << EOT | tee "$shim_file"
 #!/bin/bash
-KATA_CONF_FILE=/opt/kata/share/defaults/kata-containers/configuration.toml /opt/kata/bin/${shim_binary} \$@
+KATA_CONF_FILE=/opt/kata/share/defaults/kata-containers/configuration-${shim}.toml /opt/kata/bin/containerd-shim-kata-v2 \$@
```

Comment:
Shouldn't we be calling the respective shim binary file here instead of `/opt/kata/bin/containerd-shim-kata-v2` always?

Reply:
There's a bit of indirection here.  We are populating the 'respective shim binary file' here, actually.  This is the resulting /usr/local/bin/containerd-shim-kata-*-v2, which actually calls the standard one, but with its own unique configuration.toml.